### PR TITLE
ComputeNode subclass that will defer updates if its value did not change

### PR DESCRIPTION
**Pull request by hppeng-wynn**
ValueCheckComputeNode: checks the value of its computed function before deciding to send state updates

possibly do this for all item slots but its kinda doom rn since update items does not exist

### CODE REVIEW

File: js/atree.js [hunk 1]
```diff
@@ -799,6 +799,7 @@ function render_AT(UI_elem, list_elem, tree) {
 
         node_elem.addEventListener('click', function(e) {
             if (e.target !== this && e.target!== this.children[0]) {return;}
+            console.log("???");
```
Comment (ferricles) [inline js/atree.js:802]: delete

File: js/computation_graph.js [hunk 2]
```diff
@@ -14,7 +14,10 @@ class ComputeNode {
         this.name = name;
         this.update_task = null;
         this.fail_cb = false;   // Set to true to force updates even if parent failed.
-        this.dirty = true;
+        this.dirty = 2;         // 3 states:
+                                // 2: dirty
+                                // 1: possibly dirty
+                                // 0: clean
```
Comment (ferricles) [inline js/computation_graph.js:20]: not 100% necessary but does js has enums

worth using them over int values?

Comment (ferricles) [reply js/computation_graph.js:20]: js does have enums*

**Review by ferricles — CHANGES_REQUESTED** [review]
lgtm for most part

File: js/computation_graph.js (hunk 2 above)
Comment (hppeng-wynn) [reply js/computation_graph.js:20]: sad

File: js/atree.js (hunk 1 above)
Comment (hppeng-wynn) [reply js/atree.js]: done

**Pull request merged**